src/backend/executor/execQual.c

```diff
@@ -2294,6 +2290,9 @@ ExecMakeTableFunctionResult(ExprState *funcexpr,
 		FuncExprState *fcache = (FuncExprState *) funcexpr;
 		ExprDoneCond argDone;
 
+		funcrettype = exprType((Node *) funcexpr->expr);
```

Comment:
funcexpr also is dereferenced [below](https://github.com/arenadata/gpdb/pull/761/files#diff-924cc2c9527218d9cb8b9606e381bd58ff3a1f74fbf20f65b95c0cf1b525790eR2401)